File: backend/app/schemas/attachment.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AttachmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    page_id: int
    uploaded_by_id: int
    original_filename: str
    stored_filename: str
    s3_key: str
    content_type: str
    size_bytes: int
    created_at: datetime


class AttachmentDownloadUrlResponse(BaseModel):
    url: str
    expires_in: int

File: frontend/src/types/attachment.ts
export interface Attachment {
  id: number;
  workspace_id: number;
  page_id: number;
  uploaded_by_id: number;
  original_filename: string;
  stored_filename: string;
  s3_key: string;
  content_type: string;
  size_bytes: number;
  created_at: string;
}

export interface AttachmentDownloadUrl {
  url: string;
  expires_in: number;
}
